man/man8/zfs-receive.8

```diff
@@ -379,6 +384,22 @@ restrictions (e.g. set-once) apply equally to
 Abort an interrupted
 .Nm zfs Cm receive Fl s ,
 deleting its saved partially received state.
+.It Xo
+.Nm
+.Cm receive
+.Fl c
+.Op Fl vn
+.Ar snapshot
+.Xc
+Attempt to correct data corruption in the specified
+.Nm snapshot,
+by using the provided stream as the source of healthy data. This method of
+healing can only heal data blocks present in the stream. Metadata can not
+be healed by corrective receive. Running a scrub is recommended post
```

Comment:
That's disappointing. Why can metadata not be healed in this way? Is there any other existing way it can be healed or could that facility be added to this PR?

Reply:
`The other thing to point out is that we can only correct data. Specifically, we are only able to heal records of type DRR_WRITE and DRR_SPILL since those are the only ones that contain all of the data needed to recreate the damaged block.` from the body of the PR. I guess my real question is "how can I correct these "small errors" without a full disk read worth of scrub times two? But it doesn't belong here so I'll make a new issue to describe my situation and link to other similar things I found last night while unable to sleep.